packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_CoarseOperator_def.hpp

```diff
@@ -317,10 +364,58 @@ namespace FROSch {
             CoarseSolveComm_ = this->MpiComm_->split(!OnCoarseSolveComm_,this->MpiComm_->getRank());
             CoarseSolveMap_ = Xpetra::MapFactory<LO,GO,NO>::Build(CoarseMap_->lib(),-1,tmpCoarseMap->getNodeElementList(),0,CoarseSolveComm_);
             
+        }
+        else if(!DistributionList_->get("Type","linear").compare("Zoltan2")){
+            Teuchos::RCP<Teuchos::FancyOStream> fancy = Teuchos::fancyOStream(Teuchos::rcpFromRef(std::cout));
+
+            GatheringMaps_.resize(1);
+            CoarseSolveExporters_.resize(1);
+
+            GatheringMaps_[0] = Teuchos::rcp_const_cast<Map> (BuildUniqueMap(k0->getRowMap()));
+            //
+            CoarseSolveExporters_[0] = Xpetra::ExportFactory<LO,GO,NO>::Build(CoarseMap_,GatheringMaps_[0]);
+            
+            CrsMatrixPtr k0Unique = Xpetra::MatrixFactory<SC,LO,GO,NO>::Build(GatheringMaps_[0],k0->getGlobalMaxNumRowEntries());
+            
+            k0Unique->doExport(*k0,*CoarseSolveExporters_[0],Xpetra::INSERT);
+            k0Unique->fillComplete(GatheringMaps_[0],GatheringMaps_[0]);
+            if (NumProcsCoarseSolve_<this->MpiComm_->getSize()) {
+                ParameterListPtr tmpList = sublist(DistributionList_,"Zoltan2 Parameter");
+                ParameterListPtr zoltanParameterList = Teuchos::rcp(new ParameterList("Zoltan2 Parameter"));
+                
+                zoltanParameterList->set("algorithm", tmpList->get("algorithm","parmetis"));
+                zoltanParameterList->set("debug_level", tmpList->get("debug_level","basic_status"));
+                zoltanParameterList->set("debug_procs", tmpList->get("debug_procs","0"));
+                zoltanParameterList->set("error_check_level", tmpList->get("error_check_level","basic_assertions"));
+                zoltanParameterList->set("imbalance_tolerance",tmpList->get("imbalance_tolerance",1.1));
+                zoltanParameterList->set("num_global_parts", NumProcsCoarseSolve_);
```

Comment:
Why don't we directly use `tmpList`?

Reply:
fixed, previously the Zoltan-FROSch-Parameterlist had other/additional parameters, which were  converted to parameters permitted by Zoltan.